src/atlas/array/helpers/ArrayForEach.h

```diff
@@ -188,6 +194,12 @@ struct ArrayForEachImpl<ExecutionPolicy, Dim, ItrDim, ItrDims...> {
               tuplePushBack(slicerArgs, idx),
               tuplePushBack(maskArgs, idx));
         }
+        else if constexpr(std::is_same_v<ExecutionPolicy, execution::parallel_policy>) {
```

Comment:
Perhaps rather than the three `if constexptr, else if constexpr, else` blocks, we could just define a `constexpr auto demote_policy(execution_policy)` function that manages the policy changes.

(I know I put two of them there in the first place ☹)

Reply:
Done. Very good idea!